### PR TITLE
[android] Add Kayak setting: opt-in on F-Droid and an easy opt-out for other builds

### PR DESCRIPTION
**Pull request by pastk**
Adds a new setting to enable/disable Kayak.com affiliate links.
- based on #6720 

- closes https://github.com/organicmaps/organicmaps/issues/6773
- partially solves https://github.com/organicmaps/organicmaps/issues/7342

image

F-Droid flavor:
- make Kayak links opt-in / disabled by default (unless a user had already accepted the Kayak disclaimer before)
- if explicitly enabled by a user, then no need to display a disclaimer dialog for the first use
- links contain a referral part hence OM will get a referral bonus for bookings (reverts #7262)

Other flavors:
- added a button to easily disable Kayak links from the first-use disclaimer dialog

image

### CODE REVIEW

**Comment by Misalf-git** [comment]
> - if explicitly enabled by a user, then no need to display a disclaimer dialog for the first use

Maybe I don't get it.
Wouldn't the dialog, especially with its "details ..." button, still be useful for _any_ first time user?
A user might have toggled the setting on/off out of curiosity or even accidentally.

**Comment by pastk** [comment]
> 
> Maybe I don't get it. Wouldn't the dialog, especially with its "details ..." button, still be useful for _any_ first time user?

(the "Details on Kayak" button just proceeds to kayak.com link)

There is a detailed description of the setting, its more detailed than the disclaimer dialog even. So if a user have explicitly enabled it then I assume they have accepted it and there is no need to show and accept the same disclaimer second time..

> A user might have toggled the setting on/off out of curiosity or even accidentally.

Could happen, but its a niche case. We can also say that a user might have accepted the first-use disclaimer dialog accidentally / out of curiosity... But we won't add another "Are you really-really sure?" dialog, right? :)
Anyway the setting is there still and the user can return to it at any time.

**Comment by biodranik** [comment]
What about showing a "More information" button and display "Unfortunately, some FDroid folks are against displaying more detailed information about a hotel with a minor referral bonus to support Organic Maps if you book it. You can use Organic Maps from Google Play, AppGallery, or Obtainium to support the project" ?

**Comment by pastk** [comment]
> I am more inclined to remove it entirely, as it doesn't make sense in its current state.

But what about users who find this feature convenient?

**Comment by pastk** [comment]
> with a minor referral bonus to support Organic Maps if you book it. You can use Organic Maps from Google Play, AppGallery, or Obtainium to support the project" ?

Actually this PR brings F-Droid OM users a possibility to support OM via referral bookings. There was no such option since https://github.com/organicmaps/organicmaps/pull/7262

**Comment by pastk** [comment]
@organicmaps/translations please review [strings changes](https://github.com/organicmaps/organicmaps/pull/9820/commits/7821c1d453c748952c8e193163a3f814c41267dd#diff-8e9891f34f779c3f69c75ce68b56207e638f199a26123422c079734a32496c7c) and post amendments here (if any) - I'll submit changes in a separate PR.

File: data/strings/strings.txt [hunk 1]
```diff
@@ -18342,6 +18435,53 @@
...
+    ja = Kayakの無効化
+    ko = Kayak 비활성화
+    lt = Išjungti "Kayak
+    lv = Atslēgt Kayak
+    mr = Kayak अकषम कर
+    nb = Deaktiver Kayak
+    nl = Kayak uitschakelen
+    pl = Proszę wyłączyć Kayak
+    pt = Desativar Kayak
+    pt-BR = Desativar o Kayak
```
Comment (matheusgomesms) [inline data/strings/strings.txt:18472]: pt-BR = Desativar Kayak

File: data/strings/strings.txt [hunk 2]
```diff
@@ -8461,6 +8461,99 @@
...
+    nb = Bilder, anmeldelser og bestillingslenker vises for hoteller som har en oppføring på Kayak.com. Henvisningsbonusen som mottas for hver bestilling gjennom disse lenkene går til utvikling av Organic Maps.
+    nl = Foto's, beoordelingen en reserveringslinks worden weergegeven voor hotels die een vermelding op Kayak.com hebben. De verwijzingsbonus die voor elke boeking via deze links wordt ontvangen, gaat naar de ontwikkeling van Organic Maps.
+    pl = Zdjęcia, recenzje, linki do rezerwacji są wyświetlane dla hoteli, które mają wpis na Kayak.com. Premia za polecenie otrzymana za każdą rezerwację za pośrednictwem tych linków jest przeznaczona na rozwój Organic Maps.
+    pt = São apresentadas fotografias, críticas e links de reserva para os hotéis que têm uma listagem no Kayak.com. O bónus de referência recebido por cada reserva efectuada através destas ligações destina-se ao desenvolvimento do Organic Maps.
+    pt-BR = Fotos, avaliações e links de reserva são exibidos para os hotéis que estão listados no Kayak.com. O bônus de indicação recebido por cada reserva feita por meio desses links é destinado ao desenvolvimento do Organic Maps.
```
Comment (matheusgomesms) [inline data/strings/strings.txt:8544]: pt = São apresentadas fotografias, críticas e ligações de reserva para os hotéis que estão listados no Kayak.com. O bónus de referência recebido por cada reserva efetuada através destas ligações destina-se ao desenvolvimento do Organic Maps.
    pt-BR = Fotos, avaliações e links de reserva são exibidos para os hotéis que estão listados no Kayak.com. O bônus de indicação recebido por cada reserva feita por meio desses links é destinado ao desenvolvimento do Organic Maps.

File: data/strings/strings.txt [hunk 3]
```diff
@@ -8461,6 +8461,99 @@
...
+    it = Link agli hotel di Kayak.com
+    ja = Kayak.comホテルリンク
+    ko = Kayak.com 호텔 링크
+    lt = Kayak.com Viešbučių nuorodos
+    lv = Kayak.com Viesnīcu saites
+    mr = Kayak.com हटल लकस
+    nb = Kayak.com Hotellkoblinger
+    nl = Kayak.com Hotellinks
+    pl = Linki do hoteli Kayak.com
+    pt = Ligações de hotéis Kayak.com
```
Comment (matheusgomesms) [inline data/strings/strings.txt:8496]: pt = Ligações de hotéis do Kayak.com

**Review by matheusgomesms — COMMENTED** [review]
PT and PT-BR reviewed

**Review by andergd — CHANGES_REQUESTED** [review]
ES & EU revised

**Comment by pastk** [comment]
Thanks for translation fixes, I've included them into https://github.com/organicmaps/organicmaps/pull/9882

**Comment by licaon-kter** [comment]
Glad this is here now

Sad that it took >1 year and resulted in so many angry posts and toots from the team

**Pull request merged**